modules/programs/zsh/lib.nix

```diff
@@ -31,7 +31,17 @@ rec {
   # given a relative (or unknown) path, returns absolute by prepending home dir
   # if path doesn't begin with "/"
   mkAbsPathStr =
-    pathStr: cleanPathStr ((lib.optionalString (!lib.hasPrefix "/" pathStr) "${homeDir}/") + pathStr);
+    pathStr:
+    let
+      cleanPath = lib.removeSuffix "/" pathStr;
+      hasShellVars = lib.hasPrefix "$" cleanPath || lib.hasInfix "$" cleanPath;
+    in
+    if hasShellVars then
+      # Does not escape shell variables, allowing them to be expanded at runtime
+      cleanPath
```

Comment:
IMO this should issue a warning and be removed (maybe after a deprecation period). I don't see a good reason to support shell variables when Nix is used to manage the shell configuration.

Reply:
Just flexibility to support generating valid `zsh` configurations.